### PR TITLE
Added `withProject` method to set project for requests

### PR DESCRIPTION
**Pull request by ManukMinasyan**
### What:

- [ ] Bug Fix
- [x] New Feature

### Description:

Added `withProject` method to factory.

### CODE REVIEW

**Comment by Flame1994** [comment]
Hey all,

Is there anybody available to review this PR so we can use the project functionality?

Would be great for the new setup openai has.

Thanks!

File: src/Factory.php [hunk 1]
```diff
@@ -74,6 +79,16 @@ public function withOrganization(?string $organization): self
         return $this;
     }
 
+    /**
+     * Sets the project for the requests.
+     */
+    public function withProject(?string $project): self
+    {
+        $this->project = project;
```
Comment (skyferix) [inline src/Factory.php:87]: You forgot the dollar sign

File: src/OpenAI.php [hunk 2]
```diff
@@ -10,11 +10,12 @@ final class OpenAI
...
      * Creates a new Open AI Client with the given API token.
      */
-    public static function client(string $apiKey, ?string $organization = null): Client
+    public static function client(string $apiKey, ?string $organization = null,  ?string $project = null): Client
     {
         return self::factory()
             ->withApiKey($apiKey)
             ->withOrganization($organization)
+            ->withProject($project)
             ->withHttpHeader('OpenAI-Beta', 'assistants=v1')
```
Comment (metola) [inline src/OpenAI.php:19]: In this line, actual change to v2

**Comment by gehrisandro** [comment]
Hi @ManukMinasyan 

Thank you very much for your PR!

**Pull request merged**